### DISCUSSION TITLE
iOS 16 SwiftUI ImageRenderer

### DISCUSSION THREAD

**Discussion by john-flanagan** [pointfreeco/swift-snapshot-testing#612] [Ideas]
With iOS 16 Apple is introducing a class purpose built for converting SwiftUI `View`s into `UIImage`s called [ImageRenderer](https://developer.apple.com/documentation/swiftui/imagerenderer).

Seems like it could be a great tool for more reliable SwiftUI snapshot tests.

I put together a PoC branch as a starting point, but I'm hoping to start a discussion on the approach and the API design.
https://github.com/pointfreeco/swift-snapshot-testing/compare/main...john-flanagan:swift-snapshot-testing:swift-ui-image-renderer

**Comment by adozenlines** [comment]
What happens when a view in its initial state transitions to another state, for example, an overlay view in response to a changing property? Is the initial UI state captured as an image or the final UI state in response to view update?
